src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -248,7 +253,7 @@ public ScrollQueryFetchSearchResult executeScan(InternalScrollSearchRequest requ
         contextProcessing(context);
         try {
             processScroll(request, context);
-            if (context.searchType() == SearchType.COUNT) {
+            if (context.searchType() == SearchType.QUERY_THEN_FETCH) {
```

Comment:
This happens on subsequent scroll requests and we need the size to be greater than 0 since this is the number of documents that we fetch on each shard at this point.

If my understanding is correct, with the scan search type we need to execute a count on the first request in order to count the number of matches (so that totalHits is correct in the response). Then for each scroll we only iterate over the subset of documents that we need.